tests/jaxley_vs_neuron/test_cell.py

```diff
@@ -253,3 +255,181 @@ def integrate():
         [list(voltage1), list(voltage2), list(voltage3), list(voltage4)]
     )
     return voltages
+
+
+def _jaxley_ion_diffusion(
+    dt, t_max, i_delay, i_dur, i_amp, v_init, diams, lengths, r_as
+):
+    comp = jx.Compartment()
+    ncomp = 2
+    branch = jx.Branch(comp, ncomp=ncomp)
+    cell = jx.Cell(branch, parents=[-1, 0, 0])
+
+    for i in range(3):
+        cell.branch(i).set("radius", diams[i] / 2)
+        cell.branch(i).set("axial_resistivity", r_as[i])
+        cell.branch(i).set("length", lengths[i] / ncomp)
+
+    cell.set("v", v_init)
+    cell.branch(0).insert(HH())
+
+    # Define calcium dynamics.
+    cell.branch([0, 1]).insert(CaHVA())
+    cell.insert(CaFaradayConcentrationChange())
+    cell.insert(CaNernstReversal())
+    cell.branch(0).set("CaHVA_gCaHVA", 0.00001)
+    cell.branch(1).set("CaHVA_gCaHVA", 0.000005)
+    cell.set("eCa", 0.0)
+
+    cell.diffuse("CaCon_i")
+    cell.set("axial_diffusion_CaCon_i", 1.0)
+
+    cell.init_states()
+
+    # Stimulate.
+    step_current = jx.step_current(i_delay, i_dur, i_amp, dt, t_max)
+    cell.branch(0).comp(0).stimulate(step_current)
+
+    # Recordings of voltage and cacon_i at different compartments.
+    cell.branch(0).comp(0).record("v")
+    cell.branch(1).comp(1).record("v")
+    cell.branch(2).comp(0).record("v")
+    cell.branch(0).comp(1).record("CaCon_i")
+    cell.branch(1).comp(0).record("CaCon_i")
+    cell.branch(2).comp(1).record("CaCon_i")
+
+    recordings = jx.integrate(cell, t_max=t_max, delta_t=dt)
+    return recordings
+
+
+def _neuron_ion_diffusion(
+    dt, t_max, i_delay, i_dur, i_amp, v_init, diams, lengths, r_as
+):
+    dirname = os.path.dirname(__file__)
+    fname = os.path.join(dirname, "../arm64/.libs/libnrnmech.so")
+    h.nrn_load_dll(fname)
+
+    h.secondorder = 0
+    h.dt = dt
+
+    for sec in h.allsec():
+        h.delete_section(sec=sec)
+
+    branch0 = h.Section()
+    branch1 = h.Section()
+    branch2 = h.Section()
+
+    branch0.insert("hh")
+
+    for i, branch in enumerate([branch0, branch1, branch2]):
+        branch.nseg = 2
+        branch.Ra = r_as[i]
+        branch.L = lengths[i]
+        branch.diam = diams[i]
+
+    branch0.insert("Ca_HVA")
+    branch1.insert("Ca_HVA")
+    branch0.gCa_HVAbar_Ca_HVA = 0.00001
+    branch1.gCa_HVAbar_Ca_HVA = 0.000005
+
+    branch1.connect(branch0, 1, 0)
+    branch2.connect(branch0, 1, 0)
+
+    # Set diffusion to zero. This is causing the different interpretation of `.area()`
+    # of NEURON and therefore causes this test to fail (also for d>0) if the radiuses
+    # of compartments are not the same (as is the case here).
```

Comment:
We could also make this a test and xfail it if we want

Reply:
no because it will cause a segmentation fault later on :D